### PR TITLE
Make rule schema a bit nicer

### PR DESCRIPTION
**Pull request by underyx**
Mainly adds titles, but also simplifies some `anyOf` stuff to use `oneOf` instead.

### CODE REVIEW

File: semgrep/semgrep/rule_schema.yaml [hunk 1]
```diff
@@ -127,6 +135,7 @@ definitions:
     type: object
     properties:
       pattern-not:
+        title: (deprecated, use pattern-not-inside) Do not return finding where Semgrep pattern matches exactly
```
Comment (mschwager) [inline semgrep/semgrep/rule_schema.yaml]: Is this deprecated? I don't think we have that language anywhere else

Comment (underyx) [reply semgrep/semgrep/rule_schema.yaml]: Discussed many times before, thought I'd just document it as such and let a discussion start here :D

Comment (underyx) [reply semgrep/semgrep/rule_schema.yaml]: cc @dlukeomalley

Comment (mschwager) [reply semgrep/semgrep/rule_schema.yaml]: I suspect that discussion will greatly slow down these unrelated changes. We'll probably need to figure out a deprecation plan, docs updates, etc, if we do in fact want to deprecate `pattern-not`. If you want to move quickly with these changes it'd probably be easiest to keep `pattern-not` deprecation separate :man_shrugging:

File: semgrep/semgrep/rule_schema.yaml [hunk 2]
```diff
@@ -3,6 +3,7 @@ $schema: http://json-schema.org/draft-07/schema#
 definitions:
   patterns-content:
     type: array
+    title: "Return finding where all of the nested conditions are true"
```
Comment (mschwager) [inline semgrep/semgrep/rule_schema.yaml:6]: Nit: this one has quotes for `title` and the others don't, not sure if this is intentional

File: semgrep/semgrep/rule_schema.yaml [hunk 3]
```diff
@@ -71,6 +72,7 @@ definitions:
     type: object
     properties:
       pattern:
+        title: Return finding where Semgrep pattern matches exactly
```
Comment (mschwager) [inline semgrep/semgrep/rule_schema.yaml:75]: Another nit: we could ape the descriptions from the [docs schema table](https://semgrep.dev/docs/writing-rules/rule-syntax/#schema) if that's easier/more consistent

**Review by mschwager — APPROVED** [review]

**Comment by aryx** [comment]
ok merging then, it's been a while.

**Pull request merged**